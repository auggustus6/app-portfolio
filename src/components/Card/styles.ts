import styled from 'styled-components/native';
import { Dimensions } from 'react-native';
import { RFValue } from 'react-native-responsive-fontsize';

const { width } = Dimensions.get('window');

export const Container = styled.View.attrs({
  elevation: 5,
  shadowColor: '#000',
  shadowOffset: {
    width: 0,
    height: 5,
  },
  shadowOpacity: 0.25,
  shadowRadius: 3.84,
  marginVertical: 10,
  marginHorizontal: 10,
})`
  ${({ theme }) => `
      background:${theme.colors.white};
      width:${width * 0.9}px;
      border-radius:${theme.border.radius};
    `}
`;

export const BackgroundImage = styled.ImageBackground.attrs({
  borderTopLeftRadius: 8,
  borderTopRightRadius: 8,
})`
  ${({ theme }) => `
      height:${RFValue(180)}px;
      width:100%;
      align-items:flex-end;
      padding:${theme.spacings.small};
    `};
`;

export const ContainerInfos = styled.View`
  ${({ theme }) => `
      padding:${theme.spacings.small};
    `}
`;

export const Text = styled.Text`
  ${({ theme }) => `
      font-size:${RFValue(16)}px;
      color:${theme.colors.gray};
      font-family:${theme.font.family};
      padding-left: ${theme.spacings.small};

    `}
`;

export const Price = styled.Text`
  ${({ theme }) => `
      font-size:${RFValue(22)}px;
      color:${theme.colors.primary};
      font-family:${theme.font.familyBold};
      padding-left: ${theme.spacings.small};
    `}
`;

export const Separator = styled.View`
  ${({ theme }) => `
      width:100%;
      height:1px;
      background:${theme.colors.lightGray};
      margin:${theme.spacings.small} 0;
    `}
`;

export const GridOptions = styled.View`
  ${({ theme }) => `
      flex-direction:row;
      flex-wrap:wrap;
      margin-top:${theme.spacings.small};
    `}
`;

export const ContainerOptions = styled.View`
  ${() => `
      align-items:center;
      justify-content:center;
      flex:1;
    `}
`;

export const Label = styled.Text`
  ${({ theme }) => `
      font-size:${RFValue(12)}px;
      color:${theme.colors.gray};
      font-family:${theme.font.family};
      margin-right: ${theme.spacings.small};
    `}
`;
