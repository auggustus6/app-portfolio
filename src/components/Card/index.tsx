import React from 'react';
import { useTheme } from 'styled-components';
import { RFValue } from 'react-native-responsive-fontsize';
import { formatNumber } from '../../utils/formatedCurrency';
import Icon from 'react-native-vector-icons/MaterialIcons';
import IconEvil from 'react-native-vector-icons/EvilIcons';
import * as Styles from './styles';

type ImageSource = {
  uri: string;
};

export type CardProps = {
  label: string;
  imageSource: ImageSource;
  price: number;
  bathrooms: number;
  bedrooms: number;
  parkingSpaces: number;
};

type CardOptions = {
  label: string;
  iconName: string;
};

const Card = ({
  label,
  imageSource,
  price,
  bathrooms,
  bedrooms,
  parkingSpaces,
}: CardProps) => {
  const theme = useTheme();

  const CardOptions = ({ label, iconName }: CardOptions) => {
    return (
      <Styles.ContainerOptions>
        <Icon name={iconName} size={22} color={theme?.colors.lightGray} />
        <Styles.Label>{label}</Styles.Label>
      </Styles.ContainerOptions>
    );
  };

  return (
    <Styles.Container testID="card.container">
      <Styles.BackgroundImage
        testID="card.background"
        source={imageSource}
        resizeMode="cover">
        <IconEvil name="heart" size={RFValue(42)} color={theme?.colors.white} />
      </Styles.BackgroundImage>
      <Styles.ContainerInfos>
        <Styles.Price>R$ {formatNumber(price)}</Styles.Price>
        <Styles.Text>{label}</Styles.Text>

        <Styles.Separator />

        <Styles.GridOptions>
          <CardOptions label={`${bedrooms} Quarto(s)`} iconName="king-bed" />
          <CardOptions label={`${bathrooms} Banheiro(s)`} iconName="bathtub" />
          <CardOptions
            label={`${parkingSpaces} Vaga(s)`}
            iconName="directions-car"
          />
        </Styles.GridOptions>
      </Styles.ContainerInfos>
    </Styles.Container>
  );
};

export default Card;
