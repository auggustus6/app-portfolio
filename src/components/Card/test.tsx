/* eslint-disable no-undef */

import React from 'react';
import Card from '.';
import { CardProps } from '.';
import { render } from '../../utils/tests/helpers';

const cardProps: CardProps = {
  label: 'Restaurantes',
  imageSource: require('../../assets/images/header.png'),
  index: 0,
};

it('should render correctly', () => {
  const { toJSON } = render(<Card {...cardProps} />);
  expect(toJSON()).toMatchSnapshot();
});

it('should render label', () => {
  const { getByText } = render(<Card {...cardProps} />);
  const text = getByText(/Restaurantes/i);
  expect(text).toBeDefined();
});

it('should render some image source', async () => {
  const { getByTestId } = render(<Card {...cardProps} />);
  const imageElement = getByTestId('card.background');

  expect(imageElement.props.source).toBe(cardProps.imageSource);
});

it('should add margin left if index is odd', async () => {
  const { getByTestId } = render(<Card {...cardProps} index={1} />);
  const container = getByTestId('card.container');

  expect(container).toHaveStyleRule('margin-left', 20);
});
