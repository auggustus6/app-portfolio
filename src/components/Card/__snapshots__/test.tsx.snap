// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<View
  elevation={5}
  marginHorizontal={10}
  marginVertical={10}
  shadowColor="#000"
  shadowOffset={
    Object {
      "height": 5,
      "width": 0,
    }
  }
  shadowOpacity={0.25}
  shadowRadius={3.84}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "height": 337.5,
        "width": 337.5,
      },
    ]
  }
  testID="card.container"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "height": "100%",
          "justifyContent": "flex-end",
          "width": "100%",
        },
      ]
    }
  >
    <Image
      borderRadius={8}
      resizeMode="cover"
      source={
        Object {
          "testUri": "../../../src/assets/images/header.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="card.background"
    />
    <Text
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "Poppins-Regular",
            "fontSize": 14,
            "fontWeight": "bold",
            "paddingBottom": 8,
            "paddingLeft": 8,
          },
        ]
      }
    >
      Restaurantes
    </Text>
  </View>
</View>
`;
